internal/pkg/runtime/engine/apptainer/container_linux.go

```diff
@@ -1817,6 +1827,17 @@ func (c *container) addBindsMount(system *mount.System) error {
 		bindOpt := ""
 		if src == localtimePath || src == hostsPath {
 			bindOpt = "skip-on-error"
+			if c.engine.EngineConfig.GetSessionLayer() == apptainer.OverlayLayer {
```

Comment:
Inserting this code here interrupts the flow for setting and using bindOpt.  I think it would be better to separately check for `src == localtimePath` instead of nesting it here.

Reply:
thanks! updated in the latest commit.